src/open-id-connect-navigation-strategies.ts

```diff
@@ -22,7 +22,14 @@ export default class OpenIdConnectNavigationStrategies {
   public async signInRedirectCallback(instruction: NavigationInstruction): Promise<any> {
     const callbackHandler = async () => {
       const args: any = {};
-      return this.userManager.signinRedirectCallback(args);
+      return this.userManager.signinRedirectCallback(args).then((user) => {
+        // The state is not persisted with the rest of the user.
+        // The resolve callback is the only place you will be able to capture that state.
+        if (user.state) {
+          // When the state is set, redirect to this request route.
+          this.$window.location.href = user.state;
+        }
```

Comment:
If I am not mistaken, if we pass `const args = { redirect_url: loginRedirectRoute }` to `userManager.signinRedirect(...)`, then we will not need to do this additional redirection.

Reply:
Is so, that would be nice.
My starting point was also the issue referenced by @mttmccb.

Reply:
I don't think ```redirect_url``` works, we need the additional state param added to the ```signinRedirect```.
The ```redirect_url``` is the one which was configured on the security token service (STS) server.